File: lee/src/main/java/org/example/lee/题目/二叉树/b_二叉树的右视图.go
package 二叉树

import "container/list"

func rightSideView(root *TreeNode) []int {
	l := list.New()
	l.PushFront(root)
	var arr []int
	if root == nil {
		return arr
	}
	for l.Len() > 0 {
		length := l.Len()
		var a *TreeNode
		for i := 0; i < length; i++ {
			a = l.Front().Value.(*TreeNode)
			if a.Left != nil {
				l.PushBack(a.Left)
			}
			if a.Right != nil {
				l.PushBack(a.Right)
			}
			l.Remove(l.Front())
		}
		arr = append(arr, a.Val)
	}
	return arr
}

File: lee/src/main/java/org/example/lee/题目/二叉树/b_二叉树的右视图.java
package org.example.lee.题目.二叉树;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

import org.example.lee.model.TreeNode;

public class b_二叉树的右视图 {

	public List<Integer> rightSideView(TreeNode root) {
		List<Integer> list = new ArrayList<>();
		Queue<TreeNode> queue = new LinkedList<>();
		if (root == null) {
			return list;
		}
		queue.add(root);
		while (!queue.isEmpty()) {
			int size = queue.size();
			TreeNode poll = null;
			for (int i = 0; i < size; i++) {
				poll = queue.poll();
				if (poll.left != null) {
					queue.add(poll.left);
				}
				if (poll.right != null) {
					queue.add(poll.right);
				}
			}
			list.add(poll.val);
		}
		return list;
	}
}
